server/methods/catalog.js

```diff
@@ -700,7 +700,7 @@ Meteor.methods({
   "products/deleteProduct": function (productId) {
     check(productId, Match.OneOf(Array, String));
     // must have admin permission to delete
-    if (!Reaction.hasAdminAccess()) {
+    if (!Reaction.hasPermission("createProduct") && !Reaction.hasAdminAccess()) {
```

Comment:
I think this should be an `||`?   If you have `createProduct` OR `hasAdminAccess`?